File: leetcode_75/two_pointers/0283.java
class Solution {
    public void moveZeroes(int[] nums) {
        Queue<Integer> numsSaved = new LinkedList<>();

        for(int i = 0; i < nums.length; i++){
            int element = nums[i];

            if(element != 0){
                numsSaved.add(element);
            }
        }

        for(int i = 0; i < nums.length; i++){
            if(!numsSaved.isEmpty()){
                nums[i] = numsSaved.remove();
            }else{
                nums[i] = 0;
            }
        }
    }
}

File: snap_prep/0283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        non_zero = []
        for i in range(len(nums)):
            if nums[i] != 0:
                non_zero.append(nums[i])
        
        for i in range(len(nums)):
            if i < len(non_zero):
                nums[i] = non_zero[i]
            else:
                nums[i] = 0
        
